applications/GeoMechanicsApplication/tests/cpp_tests/test_newton_raphson_strategy_linear_elastic.cpp

```diff
@@ -0,0 +1,301 @@
+                pLinearSolver, 0.25, 0.5, CalculateInitialSecondDerivative);
+        auto pConvergenceCriteria = std::make_shared<DisplacementCriteria<SparseSpaceType, LocalSpaceType>>(
+            RelativeTollerance, AbsoluteTollerance);
+        auto dummy_parameters = Parameters();
+
+        return GeoMechanicNewtonRaphsonStrategyLinearElasticDynamic<SparseSpaceType, LocalSpaceType, LinearSolverType>(
+            rModelPart, pScheme, pLinearSolver, pConvergenceCriteria, pBuilderAndSolver,
+            dummy_parameters, beta, gamma, 30, false, false);
+    }
+};
+
+/// <summary>
+/// Runs the tests for NewtonRaphsonLinearElasticDynamic
+/// </summary>
+/// <param name="RelativeConvergenceTollerance"> relative tollerance for the displacement criteria</param>
+/// <param name="AbsoluteConvergenceTollerance"> absolute tollerance for the displacement criteria</param>
+/// <param name="DeltaTime"> time step size</param>
+/// <param name="CalculateInitialAcceleration">When true, the initial value for the second derivative is calculated which
+/// fullfills equilibrium in the very first timestep</param>
+/// <param name="UseDirectSolver"> When true, the sparse lu linear solver is used, when false, the cg linear solver is used </param>
+/// <param name="rExpectedDisplacementX"> expected displacements in x direction</param>
+/// <param name="rExpectedDisplacementY"> expected dusplacements in y direction</param>
+void TestNewtonRaphsonLinearElasticDynamic(double                     RelativeConvergenceTollerance,
+                                           double                     AbsoluteConvergenceTollerance,
+                                           double                     DeltaTime,
+                                           bool                       CalculateInitialAcceleration,
+                                           bool                       UseDirectSolver,
+                                           const std::vector<double>& rExpectedDisplacementX,
+                                           const std::vector<double>& rExpectedDisplacementY)
+{
+    NewtonRaphsonStrategyLinearElasticDynamicTester tester;
+
+    // set up model and solver
+    ModelPart& model_part = tester.GetModelPart();
+
+    auto& r_current_process_info = model_part.GetProcessInfo();
+
+    r_current_process_info[TIME]       = 0.0;
+    r_current_process_info[DELTA_TIME] = DeltaTime;
+
+    auto stiffness_matrix  = Matrix(2, 2);
+    stiffness_matrix(0, 0) = 6.0;
+    stiffness_matrix(0, 1) = -2.0;
+    stiffness_matrix(1, 0) = -2.0;
+    stiffness_matrix(1, 1) = 4.0;
+
+    auto mass_matrix  = Matrix(2, 2);
+    mass_matrix(0, 0) = 2.0;
+    mass_matrix(0, 1) = 0.0;
+    mass_matrix(1, 0) = 0.0;
+    mass_matrix(1, 1) = 1.0;
+
+    auto damping_matrix  = Matrix(2, 2);
+    damping_matrix(0, 0) = 0.0;
+    damping_matrix(0, 1) = 0.0;
+    damping_matrix(1, 0) = 0.0;
+    damping_matrix(1, 1) = 0.0;
```

Comment:
Since we want a zero matrix here, I'd use:
```suggestion
    auto damping_matrix  = Matrix(ZeroMatrix(2, 2));
```